libraries/AC_Avoidance/AP_OADatabase.cpp

```diff
@@ -101,45 +127,22 @@ void AP_OADatabase::update()
     }
 
     process_queue();
-    optimize_db_filter();
     database_items_remove_all_expired();
 }
 
 // push a location into the database
-void AP_OADatabase::queue_push(const Location &loc, const uint32_t timestamp_ms, const float distance, const float angle)
+void AP_OADatabase::queue_push(const Vector2f &pos, uint32_t timestamp_ms, float distance)
 {
     if (!healthy()) {
         return;
     }
 
-    AP_OADatabase::OA_DbItemImportance importance = AP_OADatabase::OA_DbItemImportance::Normal;
-
-    if (distance <= 0 || angle < 0 || angle > 360) {
-        // sanity check
-        importance = AP_OADatabase::OA_DbItemImportance::Normal;
-
-    } else if (distance < 10 && (angle > (360-10) || angle < 10)) {
-        // far and directly in front +/- 10deg
-        importance = AP_OADatabase::OA_DbItemImportance::High;
-    } else if (distance < 5 && (angle > (360-30) || angle < 30)) {
-        // kinda far and forward of us +/- 30deg
-        importance = AP_OADatabase::OA_DbItemImportance::High;
-    } else if (distance < 3 && (angle > (360-90) || angle < 90)) {
-        // near and ahead +/- 90deg
-        importance = AP_OADatabase::OA_DbItemImportance::High;
-    } else if (distance < 1.5) {
-        // very close anywhere
-        importance = AP_OADatabase::OA_DbItemImportance::High;
-
-    } else if  (distance >= 10) {
-        // really far away
-        importance = AP_OADatabase::OA_DbItemImportance::Low;
-    } else if (distance < 5 && (angle <= (360-90) || angle >= 90)) {
-        // kinda far and behind us
-        importance = AP_OADatabase::OA_DbItemImportance::Low;
+    // ignore objects that are far away
+    if ((_dist_max > 0.0f) && (distance > _dist_max)) {
+        return;
     }
 
-    const OA_DbItem item = {loc, timestamp_ms, 0, 0, importance};
+    const OA_DbItem item = {pos, timestamp_ms, MAX(_radius_min, distance * dist_to_radius_scalar), 0, AP_OADatabase::OA_DbItemImportance::Normal};
```

Comment:
forcing everything to importance::Normal will defeat the whole GCS throttling scheme. Setting the Imporantance is still needed, maybe it's just the algorithm that assigns it that needs to change?

Reply:
@magicrub,

Yes, I think if we want to try and reduce the number of objects we send to the GCS then improving the algorithm is a good idea and I guess that's what I was trying to do.  While trying to get Bendy Ruler working for Copter I found the display of the objects on the GCS was so unintelligible ([see pic in this issue](https://github.com/ArduPilot/ardupilot/issues/12661)) that I figured it was best to just simplify it down to a point where it worked and then we can re-build from there.

I didn't think that using the vehicle' heading (especially for Copter) was a good idea so what I've done is add the OA_DB_DIST_MAX parameter so the user can specify the maximum distance from the vehicle for any new objects entering the database.  This reduces the number of objects overall both for sending to the GCS but also for the path planning done in BendyRuler.